src/lem/path.rs

```diff
@@ -111,77 +105,74 @@ impl LEMCTL {
         &self,
         path: &Path,
         // `map` keeps track of the updated names of variables
-        map: &mut HashMap<Var, Var>, // name -> path.name
-    ) -> Result<Self> {
+        map: &mut VarMap<Var>, // name -> path.name
+    ) -> Self {
         match self {
             LEMCTL::MatchTag(var, cases) => {
                 let mut new_cases = vec![];
                 for (tag, case) in cases {
-                    let new_case = case.deconflict(&path.push_tag(tag), &mut map.clone())?;
+                    let new_case = case.deconflict(&path.push_tag(tag), &mut map.clone());
                     new_cases.push((*tag, new_case));
                 }
-                Ok(LEMCTL::MatchTag(
-                    retrieve_one(map, var)?,
-                    IndexMap::from_iter(new_cases),
-                ))
+                LEMCTL::MatchTag(retrieve_one(map, var), IndexMap::from_iter(new_cases))
             }
             LEMCTL::MatchSymbol(var, cases, def) => {
                 let mut new_cases = vec![];
                 for (symbol, case) in cases {
-                    let new_case = case.deconflict(&path.push_symbol(symbol), &mut map.clone())?;
+                    let new_case = case.deconflict(&path.push_symbol(symbol), &mut map.clone());
                     new_cases.push((symbol.clone(), new_case));
                 }
-                Ok(LEMCTL::MatchSymbol(
-                    retrieve_one(map, var)?,
+                LEMCTL::MatchSymbol(
+                    retrieve_one(map, var),
                     IndexMap::from_iter(new_cases),
-                    Box::new(def.deconflict(&path.push_default(), &mut map.clone())?),
-                ))
+                    Box::new(def.deconflict(&path.push_default(), &mut map.clone())),
+                )
             }
             LEMCTL::Seq(ops, rest) => {
                 let mut new_ops = vec![];
                 for op in ops {
```

Comment:
Instead of 
```rust
let mut new_ops = vec![];
for op in ops {
  match op {
    LEMOP::SomeVariant(preimg, img) => {
      let preimg = …
      let img = …
      new_ops.push(LEMOP::SomeVariant(preimg, img))
    },
    …
  }
```

You could simply write:
```rust
let new_ops = ops.into_iter().map{|op| {
  LEMOP::SomeVariant(preimg, img) => {
    let preimg = …
    let img = …
    LEMOP::SomeVariant(preimg, img)
  },
  … // same refactoring for the others
};
```

Reply:
Here we have `?`s in the middle of the code. I think the `for` loop is readable enough in this case